### PR TITLE
refactor(os-carp-vip-dhcp): shared keeper.conf shell parser + reader conformance test

### PR DESCRIPTION
**Pull request by toreamun**
## What

Follow-up to #74 (name-keyed keeper.conf): a behavioural conformance harness for the keeper.conf readers, plus the small refactor that makes it possible.

### Shared shell parser
The rc.d service and the CARP status hook each carried their own copy of the keeper.conf peel-loop. This extracts it into `keeperconf.sh`, sourced by both, so the two shell readers now parse through one implementation and cannot drift from each other. The parse logic is unchanged.

### Conformance test
`test_reader_conformance.py` runs the shared shell parser under `sh` over edge-case records (reordered keys, empty values, a value containing `=`, leading-dash values) and asserts it extracts exactly what the Python reader (`keeper_records`) does, field for field. This catches a future edit to one reader's format handling that the text-based key-name agreement test (added in #74) would not, because it exercises the real shell code, not just the key names. The key-name agreement test now points at `keeperconf.sh`, the single shell reader.

The two PHP readers only consume `request`, still covered by the key-name agreement test.

## Testing
272 unit tests, flake8, pylint 10/10, pyright 0, and `sh -n` on all three shell files, all green. shellcheck runs in CI.

## Version
Bumped to 1.13.1 (internal refactor plus test, no behaviour change).

### CODE REVIEW

File: net/os-carp-vip-dhcp/src/opnsense/scripts/OPNsense/CarpVipDhcp/keeperconf.sh [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            arplistenpromisc=*) arplistenpromisc="${_field#*=}" ;;
+            defaultroutemode=*) defaultroutemode="${_field#*=}" ;;
+            backupegress=*) backupegress="${_field#*=}" ;;
+            backupegressform=*) backupegressform="${_field#*=}" ;;
+            backupegressgateway=*) backupegressgw="${_field#*=}" ;;
+            backupegressinterface=*) backupegressiface="${_field#*=}" ;;
+            backupegressprefixes=*) backupegressprefixes="${_field#*=}" ;;
+        esac
+    done
+}
```
[the remark this answers is not in the document]
Comment (toreamun) [reply net/os-carp-vip-dhcp/src/opnsense/scripts/OPNsense/CarpVipDhcp/keeperconf.sh]: Fixed in 267032f: the scratch temporaries are now namespaced (`_kc_rec`/`_kc_field`) and `unset` at the end, so sourcing the parser no longer leaks them into the caller. Good catch.

File: net/os-carp-vip-dhcp/tests/test_reader_conformance.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+def _sh_parse(line):
+    """{key: value} as keeperconf.sh's carpvipdhcp_parse_line extracts it (one
+    `printf` per key, in _KEYS order, so the split lines up)."""
+    dump = "\n".join(f'printf "%s\\n" "${{{_VAR[k]}}}"' for k in _KEYS)
+    script = f'. "$1"\ncarpvipdhcp_parse_line "$2"\n{dump}\n'
+    out = subprocess.run(
+        ["sh", "-c", script, "sh", _KEEPERCONF_SH, line],
+        capture_output=True, text=True, check=True).stdout.split("\n")
+    return dict(zip(_KEYS, out))
```
[the remark this answers is not in the document]
Comment (toreamun) [reply net/os-carp-vip-dhcp/tests/test_reader_conformance.py:71]: Fixed in 267032f: switched to `splitlines()` and added an explicit line-count assertion, so stray output fails loudly instead of being silently dropped by zip().

**Comment by toreamun** [comment]
Squash-merged to main as ed474e0 (kept the commit dated to the work timeframe). Closing.
